core/src/main/scala/org/apache/spark/rdd/PipedRDD.scala

```diff
@@ -129,7 +130,7 @@ private[spark] class PipedRDD[T: ClassTag](
       override def run(): Unit = {
         val err = proc.getErrorStream
         try {
-          for (line <- Source.fromInputStream(err).getLines) {
+          for (line <- Source.fromInputStream(err)(StandardCharsets.UTF_8).getLines) {
```

Comment:
@zsxwing : I don't agree that using system default encoding would be a good idea. One can see different result on different systems which would be annoying. A user can run a binary in pipe() operator and let its stderr stream emit data with a character encoding that he/she wishes to.

@srowen : I agree that UTF-8 is a safer default. Making this configurable would be the ideal thing to do.